src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        if (operation == null) {
+            throw new IllegalArgumentException("Operation is null");
+        }
+        if (!operationHandlers.containsKey(operation)) {
+            return null;
+        }
```

Comment:
We will get a NPE (NullPointerException) in ShopServiceImpl#process method with this implementation + we don't want to process current and all next operations if we don't have an appropriate handler. 
Let's also add test for this)
```suggestion
        if (!operationHandlers.containsKey(operation)) {
            throw new IllegalArgumentException("...");
        }
```